packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,182 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the appNotification specific part of the SDK
+ */
+export namespace appNotification {
+  /**
+   * Data structure to represent appNotification information
+   */
+  export interface NotificationDisplayParam {
+    /**
+     * Notification title(maximum length: 20 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 40 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: string;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url link to the page in which the notification would direct the user to.
+     */
+    notificationActionUrl: URL;
+  }
+
+  /**
+   * Data structure with simplified parameters that would be passed across the iframe
+   */
+  export interface NotificationDisplayParamForAppHost {
+    /**
+     * Notification title(maximum length: 20 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 40 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: string;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url string type to the page in which the notification would direct the user to.
+     */
+    notificationActionUrlAsString: string;
+  }
+
+  /**
+   * This converts the notifcationActionUrl from a URL type to a string type for proper flow across the iframe
+   * @param ndParam
+   * @returns a NotificationDisplay object with a string type parameter for the notificationAtionUrl
```

Comment:
> @returns a NotificationDisplay object with a string type parameter for the notificationAtionUrl [](http://example.com/codeflow?start=4&length=96)

I'm not sure I see how this describes what is going on here?

Reply:
Alright, I'll reword this for better clarity.

Reply:
Does this sound better?

\**
* Data structure to represent appNotification information that would be sent across the iframe to the hubsdk
   * The notificationActionUrl in this data structure has been been serialized from a URL type to a string type for efficient transfer of data across the iframe
   */